stl/inc/format

```diff
@@ -2557,8 +2557,221 @@ template <_Format_supported_charT _CharT, class _Traits>
 struct formatter<basic_string_view<_CharT, _Traits>, _CharT>
     : _Formatter_base<basic_string_view<_CharT, _Traits>, _CharT, _Basic_format_arg_type::_String_type> {};
 
-using format_context  = basic_format_context<back_insert_iterator<string>, string::value_type>;
-using wformat_context = basic_format_context<back_insert_iterator<wstring>, wstring::value_type>;
+template <class _T>
+class _Fmt_buffer {
+private:
+    _T* _Ptr;
+    size_t _Size;
+    size_t _Capacity;
+
+protected:
+    _Fmt_buffer(size_t _Sz) noexcept : _Size(_Sz), _Capacity(_Sz) {}
+    _Fmt_buffer(_T* _P = nullptr, size_t _Sz = 0, size_t _Cap = 0) noexcept : _Ptr(_P), _Size(_Sz), _Capacity(_Cap) {}
+    ~_Fmt_buffer() = default;
+
+    void set(_T* _Buf_data, size_t _Buf_capacity) noexcept {
+        _Ptr  = _Buf_data;
+        _Size = _Buf_capacity;

```

Comment:
It is a bit strange that we have the name `_Buf_capacity` but set `_Size` to it. Should we also set `_Capacity` or rename the argument?

Reply:
Good catch, thanks. This was supposed to be `_Capacity` but got messed up during _Uglification.